diff --git a/dist/index.js b/dist/index.js
index 5cb1fd007d32021ad950d961dd4a0868576a2c0d..e4b6d59039ccc2fbcf1a52a6afdc2b3e2e48608e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -97,7 +97,7 @@ var PopperContent = React.forwardRef(
       align = "center",
       alignOffset = 0,
       arrowPadding = 0,
-      avoidCollisions = true,
+      avoidCollisions = false,
       collisionBoundary = [],
       collisionPadding: collisionPaddingProp = 0,
       sticky = "partial",
@@ -150,8 +150,8 @@ var PopperContent = React.forwardRef(
           apply: ({ elements, rects, availableWidth, availableHeight }) => {
             const { width: anchorWidth, height: anchorHeight } = rects.reference;
             const contentStyle = elements.floating.style;
-            contentStyle.setProperty("--radix-popper-available-width", `${availableWidth}px`);
-            contentStyle.setProperty("--radix-popper-available-height", `${availableHeight}px`);
+            contentStyle.setProperty("--radix-popper-available-width", `${Number.isFinite(availableWidth) ? availableWidth : 0}px`);
+            contentStyle.setProperty("--radix-popper-available-height", `${Number.isFinite(availableHeight) ? availableHeight : 0}px`);
             contentStyle.setProperty("--radix-popper-anchor-width", `${anchorWidth}px`);
             contentStyle.setProperty("--radix-popper-anchor-height", `${anchorHeight}px`);
           }
diff --git a/dist/index.mjs b/dist/index.mjs
index 9f84984eab84e32d20d6c052217da0e3b8374b40..962c370e04ae8baa0e700f4ab516f3042adbe72d 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -62,7 +62,7 @@ var PopperContent = React.forwardRef(
       align = "center",
       alignOffset = 0,
       arrowPadding = 0,
-      avoidCollisions = true,
+      avoidCollisions = false,
       collisionBoundary = [],
       collisionPadding: collisionPaddingProp = 0,
       sticky = "partial",
@@ -110,13 +110,13 @@ var PopperContent = React.forwardRef(
           ...detectOverflowOptions
         }),
         avoidCollisions && flip({ ...detectOverflowOptions }),
-        size({
+        avoidCollisions && size({
           ...detectOverflowOptions,
           apply: ({ elements, rects, availableWidth, availableHeight }) => {
             const { width: anchorWidth, height: anchorHeight } = rects.reference;
             const contentStyle = elements.floating.style;
-            contentStyle.setProperty("--radix-popper-available-width", `${availableWidth}px`);
-            contentStyle.setProperty("--radix-popper-available-height", `${availableHeight}px`);
+            contentStyle.setProperty("--radix-popper-available-width", `${Number.isFinite(availableWidth) ? availableWidth : 0}px`);
+            contentStyle.setProperty("--radix-popper-available-height", `${Number.isFinite(availableHeight) ? availableHeight : 0}px`);
             contentStyle.setProperty("--radix-popper-anchor-width", `${anchorWidth}px`);
             contentStyle.setProperty("--radix-popper-anchor-height", `${anchorHeight}px`);
           }
